src/scripts/main.js

```diff
@@ -1,3 +1,332 @@
 'use strict';
 
-// write code here
+const tableHeaders = document.querySelectorAll('th');
+const rowsArray = Array.from(document.querySelectorAll('tbody tr'));
+const sortDirection = {};
+
+// tableHeaders.forEach((header, index) => {
+//   sortDirection[index] = true;
+
+//   header.addEventListener('click', (e) => {
+//     const sortedRows = rowsArray.sort((rowA, rowB) => {
+//       const cellA = rowA.cells[index].textContent.trim();
+//       const cellB = rowB.cells[index].textContent.trim();
+
+//       if (index === 4) {
+//         const salaryA = parseFloat(cellA.replace(/[$,]/g, ''));
+//         const salaryB = parseFloat(cellB.replace(/[$,]/g, ''));
+
+//         return sortDirection[index] ? salaryA - salaryB : salaryB - salaryA;
+//       } else {
+//         return sortDirection[index]
+//           ? cellA.localeCompare(cellB)
+//           : cellB.localeCompare(cellA);
+//       }
+//     });
+
+//     const tbody = document.querySelector('tbody');
+
+//     tbody.innerHTML = '';
+//     sortedRows.forEach((row) => tbody.appendChild(row));
+
+//     sortDirection[index] = !sortDirection[index];
+//   });
+// });
+
+function sortRows(index, isAscending, rowsArr) {
```

Comment:
The function `sortRows` is doing more than sorting rows. It's also formatting the salary. Consider renaming this function to reflect this, or better yet, separate the concerns of sorting and formatting into two different functions.